### ISSUE TITLE
non-const getClassDef() called on aliased member. Please report as a bug.

### ISSUE THREAD

**Issue by marcalff** [doxygen/doxygen#7624]
With the following doxyfile:

```
malff@linux-8edv:bug_memberdef> pwd                                                                                                                                         
/home/malff/EXTERNAL/doxygen/bug_memberdef                                                                                                                                  
malff@linux-8edv:bug_memberdef> doxygen -x                                                                                                                                  
# Difference with default Doxyfile 1.8.18 (6c06e912338176303d1a1e041a39984ff6fd42be*)                                                                                       
OUTPUT_DIRECTORY       = build                                                                                                                                              
EXTRACT_ALL            = YES                                                                                                                                                
EXTRACT_PRIVATE        = YES                                                                                                                                                
EXTRACT_STATIC         = YES                                                                                                                                                
INPUT                  = ./string_view.h                                                                                                                                    
GENERATE_TREEVIEW      = YES       
```

and for the following input:

```
malff@linux-8edv:bug_memberdef> cat string_view.h                                                                                                                           
#ifndef MYSQL_HARNESS_STDX_STRING_VIEW_INCLUDED                                                                                                                             
#define MYSQL_HARNESS_STDX_STRING_VIEW_INCLUDED                                                                                                                             
                                                                                                                                                                            
namespace stdx {                                                                                                                                                            
                                                                                                                                                                            
inline namespace literals {                                                                                                                                                 
inline namespace string_view_literals {                                                                                                                                     
inline constexpr basic_string_view<char> operator""_sv(const char *str,
                                                       size_t len) noexcept {
  return basic_string_view<char>{str, len};
}
inline constexpr basic_string_view<wchar_t> operator""_sv(const wchar_t *str,
                                                          size_t len) noexcept {
  return basic_string_view<wchar_t>{str, len};
}
inline constexpr basic_string_view<char16_t> operator""_sv(
    const char16_t *str, size_t len) noexcept {
  return basic_string_view<char16_t>{str, len};
}
inline constexpr basic_string_view<char32_t> operator""_sv(
    const char32_t *str, size_t len) noexcept {
  return basic_string_view<char32_t>{str, len};
}
}  // namespace string_view_literals
}  // namespace literals


}  // namespace stdx

#endif
```


doxygen raises the following errors:

```
...
Generating search indices...
error: non-const getClassDef() called on aliased member. Please report as a bug.
error: non-const getNamespaceDef() called on aliased member. Please report as a bug.
error: non-const getClassDef() called on aliased member. Please report as a bug.
error: non-const getNamespaceDef() called on aliased member. Please report as a bug.
error: non-const getClassDef() called on aliased member. Please report as a bug.
error: non-const getNamespaceDef() called on aliased member. Please report as a bug.
error: non-const getClassDef() called on aliased member. Please report as a bug.
error: non-const getNamespaceDef() called on aliased member. Please report as a bug.
error: non-const getClassDef() called on aliased member. Please report as a bug.
error: non-const getNamespaceDef() called on aliased member. Please report as a bug.
error: non-const getClassDef() called on aliased member. Please report as a bug.
error: non-const getNamespaceDef() called on aliased member. Please report as a bug.
error: non-const getClassDef() called on aliased member. Please report as a bug.
error: non-const getNamespaceDef() called on aliased member. Please report as a bug.
error: non-const getClassDef() called on aliased member. Please report as a bug.
error: non-const getNamespaceDef() called on aliased member. Please report as a bug.
Generating example documentation...

```
                             
Suggested fix:

```
diff --git a/src/searchindex.cpp b/src/searchindex.cpp
index 329b1239..22c12fd9 100644
--- a/src/searchindex.cpp
+++ b/src/searchindex.cpp
@@ -1048,7 +1048,7 @@ void writeJavaScriptSearchIndex()
 
           if (dl->count()==1) // item with a unique name
           {
-            MemberDef  *md = dynamic_cast<MemberDef*>(d);
+            const MemberDef  *md = dynamic_cast<MemberDef*>(d);
             QCString anchor = d->anchor();
 
             ti << "'" << externalRef("../",d->getReference(),TRUE)
@@ -1100,7 +1100,7 @@ void writeJavaScriptSearchIndex()
               Definition *scope     = d->getOuterScope();
               Definition *next      = di.current();
               Definition *nextScope = 0;
-              MemberDef  *md        = dynamic_cast<MemberDef*>(d);
+              const MemberDef  *md        = dynamic_cast<MemberDef*>(d);
               if (next) nextScope = next->getOuterScope();
               QCString anchor = d->anchor();
```

**Comment by albert-github** [comment]
The proposed fix removes the messages and the output is identical (for the given example).

**Comment by marcalff** [comment]
Thanks for considering the fix. Please note that I only tested the HTML output.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.8.18.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
